site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2309,7 +2309,17 @@ public function getRotatingSectionsByGrader($gradeable_id) {
 
     public function getGradersByUserType() {
         $this->course_db->query(
-            "SELECT user_firstname, user_lastname, user_id, user_group FROM users WHERE user_group < 4 ORDER BY user_group, user_id ASC"
+            "SELECT 
+                (CASE WHEN user_preferred_firstname IS NOT NULL THEN user_preferred_firstname ELSE user_firstname END) as user_firstname,
+                (CASE WHEN user_preferred_lastname IS NOT NULL THEN user_preferred_lastname ELSE user_lastname END) as user_lastname,
```

Comment:
We probably also want to handle the case where `user_preferred_firstname` can be empty string?

Can do this using [`NULLIF`](https://www.postgresql.org/docs/13/functions-conditional.html#FUNCTIONS-NULLIF) function, and then also simplify the `CASE` with [`COALESCE `](https://www.postgresql.org/docs/13/functions-conditional.html#FUNCTIONS-COALESCE):

```suggestion
                COALESCE(NULLIF(user_preferred_firstname, ''), user_firstname) AS user_firstname,
                COALESCE(NULLIF(user_preferred_lastname, ''), user_lastname) AS user_lastname,
```

Reply:
That's smart, thank you!